com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/eval/JdtEvaluationProvider.java

```diff
@@ -70,15 +76,11 @@ public void initialize(IDebugAdapterContext context, Map<String, Object> props)
     @Override
     public CompletableFuture<Value> evaluate(String expression, ThreadReference thread, int depth) {
         CompletableFuture<Value> completableFuture = new CompletableFuture<>();
-        String projectName = (String) options.get(Constants.PROJECTNAME);
+
         if (debugTarget == null) {
+            String projectName = (String) options.get(Constants.PROJECTNAME);
```

Comment:
Please put the logic inside the initialize() of this provider. We only need to look for the project once per session.
Trying to calculate project everytime users evaluate expressions is overkill.